packages/flutter/test/material/ink_well_test.dart

```diff
@@ -1330,4 +1333,253 @@ void main() {
       textDirection: TextDirection.ltr,
     ));
   });
+
+  // Regression test for https://github.com/flutter/flutter/issues/6751
+  testWidgets('When InkWell/Ancestor has a GlobalKey and changes position, splash should not stop', (WidgetTester tester) async {
+    void expectTest(bool painted) {
+      final PaintPattern paintPattern = paints..circle();
+
+      expect(
+        Material.of(tester.element(find.byType(InkWell)))! as RenderBox,
+        painted ? paintPattern : isNot(paintPattern),
+      );
+    }
+
+    Future<void> test(Widget Function(Key? key, {bool onTapDownChangeWrap, bool onTapChangeWrap}) buildApp) async {
+      await tester.pumpWidget(buildApp(null, onTapChangeWrap: true));
+      await tester.tap(find.byType(InkWell));
+      await tester.pump(const Duration(milliseconds: 60));
```

Comment:
you need a pump without a duration here, before the pump with the duration, otherwise the duration will have no effect (because it'll just assume that the first frame after the tap just happened to jank hard).

same for other cases below